services/server/server.go

```diff
@@ -611,6 +618,49 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	}, nil
 }
 
+// deleteAllAlertRules deletes all alerts present rules.
+// Returns nil in case of success, otherwise an error.
+func (s *Server) deleteAllAlertRules() error {
+	alertRuleIds := s.getAlertRuleIds()
+
+	if len(alertRuleIds) > 0 {
+		err := s.db.InTransaction(func(tx *reform.TX) error {
+			return models.RemoveRules(tx.Querier, alertRuleIds)
+		})
+		if err != nil {
+			return err
+		}
+
+		s.vmalert.RequestConfigurationUpdate()
+	}
+
+	return nil
+}
+
+// getAlertRuleIds lists the alert rules ids into an array.
+// Error is not returned, when occurres an empty array is returned.
+func (s *Server) getAlertRuleIds() []string {
+	var rules []*models.Rule
+	err := s.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		rules, err = models.FindRules(tx.Querier)
+
+		return err
+	})
+
+	if err == nil {
```

Comment:
this check doesn't make sense because in Go if you return error you don't return result. 
and go usually ignores loops if you try to loop nil.

Reply:
Does not exist anymore.